NetCasbin/InternalEnforcer.cs

```diff
@@ -44,7 +45,14 @@ protected bool InternalAddPolicy(string sec, string ptype, List<string> rule)
                 return false;
             }
 
-            NoticePolicyChanged(sec);
+            if (autoBuildRoleLinks && sec.Equals(PermConstants.Section.RoleSection))
+            {
+                model.BuildIncrementalRoleLink(roleManager, PolicyOperation.PolicyAdd,
+                    sec, ptype, rule);
+                ExpressionHandler.SetGFunctions();
```

Comment:
`ExpressionHandler.SetGFunctions();` should be only set once, because the `g` function would never change, maybe we can put it in some initilization functions?

Reply:
The purpose of calling `SetGFunctions` is to clear the cache in `GFunction`. It is necessary.